make/env.mk

```diff
@@ -11,10 +11,8 @@ ifeq ($(findstring :, $(GOPATH)), $(colon))
 GOPATH := $(firstword $(subst :, ,$(GOPATH)))
 endif
 
-export CGO_ENABLED DEFAULT_GOOS GOARCH GOTAGS GO111MODULE GOBIN GOPROXY
-ifeq ($(CGO_ENABLED),)
-CGO_ENABLED := 1
```

Comment:
do we want CGO_ENABLED by default? or am I misreading this?

Reply:
I assume this question is about `make/env.mk` in general and not in the context of its use in `scanner/Makefile`, right? `make/env.mk` is used in root `Makefile` and in `tests/Makefile`.

My change here does not alter the default, it remains `1`. Except for an edge case that I described in the commit message here [`0f419b4`](https://github.com/stackrox/stackrox/pull/10853/commits/0f419b4e89d371605a2a41ffe8b2b6504d9b5ca0).

Reply:
Yeah, the question is about the general case. I didn't realize we wanted to have `CGO_ENABLED=1`, by default. I wasn't sure (still not quite sure) if enabling this by default slows down compilation even when we don't even use cgo (guess that's not a huge deal, though), and I believe we can't do cross-compilation (you mentioned this, too).

I'm also wondering if enabling cgo would even do anything since we use the upstream compiler instead of the Red Hat one. I'm definitely not an expert, but I think we'd have to use Red Hat's go compiler as well as setting `CGO_ENABLED=1` to have a FIPS compliant build.

That being said, that probably is not the goal of ROX-23969, rather the goal is to just minimize differences, and setting `CGO_ENABLED=1` is one of those differences

Reply:
> I think we'd have to use Red Hat's go compiler as well as setting `CGO_ENABLED=1` to have a FIPS compliant build

That's correct. Downstream compiler is patched in a way that it injects usage of Red Hat openssl from the base container instead of Go own cryptography. `CGO_ENABLED=1` is one of prerequisites but not the only one.

Still `CGO_ENABLED=1` on its own makes the resulting executable use system dns resolver instead of the one implemented in Go library, use glibc as dynamically loaded library (which can be an issue as I saw) and I'm sure changes behavior in many other little aspects.

Unfortunately, the compilation slowdown is the price to pay, although I don't know how big is it.

Reply:
> Still `CGO_ENABLED=1` on its own makes the resulting executable use system dns resolver instead of the one implemented in Go library

Oh I didn't know about that. Curious do you have a link to some docs which mention which things from the stdlib may be replaced with Ci implementations?

Reply:
@connorgorman mentioned DNS thing. I saw it also mentioned in StackOverflow <https://stackoverflow.com/a/61515424/484050> and <https://stackoverflow.com/a/64531740/484050> (this one also mentions a difference to user lookup).
`glibc` thing was mentioned in several places and I ran into it myself.

Reply:
BTW, I'll tag @janisz here because we have touched upon CGO some time ago.